server/eventHandlers/notifications.ts

```diff
@@ -56,10 +58,23 @@ export default class extends IEventHandler {
         event.excludeAddresses,
         event.includeAddresses
       );
-      log.trace(`Emitted ${dbNotifications.length} notifications.`);
+
+      // construct notification with all the necessary data from the DB (without having to re-query using joins)
+      const formattedEvent = dbNotification.toJSON();
+      formattedEvent.ChainEvent = dbEvent.toJSON()
+      formattedEvent.ChainEvent.ChainEventType = dbEventType.toJSON()
+
+      // publish notification to the appropriate RabbitMQ queue (sends to socket.io servers to send to clients)
```

Comment:
So -- to be clear -- we now have two different queues, one between the chain events processes and one between the server and the socket.io, used to send notifications to clients? I would love to see an architectural document.

Reply:
There is 2 queues yes: chain-events/dbNode -> consumer -> socket.io/cw_api server